platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ImportMigrationEntry.java

```diff
@@ -0,0 +1,127 @@
+ * library.
+ * </b>
+ * </p>
+ */
+public interface ImportMigrationEntry extends MigrationEntry {
+    /**
+     * Retrieves a migration entry referenced from a property value defined in the properties file
+     * associated with this migration entry.
+     * <p>
+     * The entry returned would be an entry representing the file that was referenced by the specified
+     * property value in the java properties file represented by this entry on the exported system.
+     * For example:
+     * <p>
+     * If the properties file (e.g. etc/ws-security/server/encryption.properties) represented by this
+     * entry defined the following mapping:
+     * org.apache.ws.security.crypto.merlin.x509crl.file=etc/certs/demoCA/crl/crl.pem
+     * <p>
+     * then the following code:
+     * <pre>
+     *     final ImportMigrationEntry entry
+     *         = context.getEntry("etc/ws-security/server/encryption.properties");
+     *
+     *     final Optional&lt;ImportMigrationEntry&gt; entry2
+     *         = entry.getPropertyReferenceEntry("org.apache.ws.security.crypto.merlin.x509crl.file");
+     * </pre>
+     * <p>
+     * would return an entry representing the exported file <code>etc/certs/demoCA/crl/crl.pem</code>
+     * allowing the migratable a chance to restore it in its original location and verifying that the
+     * property in the local etc/ws-security/server/encryption.properties file is still defined with
+     * the same value after the import operation has completed.
+     *
+     * @param name the name of the property that contains the reference to a migration entry
+     * @return the migration entry corresponding to the property value defined by <code>name</code>
+     * in the property file referenced by this entry or empty if it was not exported
+     * @throws IllegalArgumentException if <code>name</code> is <code>null</code>
+     */
+    public Optional<ImportMigrationEntry> getPropertyReferencedEntry(String name);
+
+    /**
+     * Stores this required entry's content appropriately based on this entry's path which can include
+     * sub-directories using the specified consumer.
+     * <p>
+     * All errors and warnings are automatically recorded with the associated migration report including
+     * those thrown by the exporter logic.
+     * <p>
+     * <i>Note:</i> The input stream will automatically be closed (if not closed already) when the
+     * operation completes successfully or not.
+     *
+     * @param consumer a consumer capable of importing the content of this entry from a provided input
+     *                 stream which might be empty if the entry was not exported otherwise an error
+     *                 will automatically be recorded)
+     * @return <code>true</code> if no errors were recorded as a result of processing this command;
+     * <code>false</code> otherwise
+     * @throws MigrationException       if a failure that prevents the operation from continuing occurred
```

Comment:
❓ Should we make the distinction between `MigrationException` being thrown and `false` being returned clearer here? When should a client expect one vs. the other? Should they care about the exception at all and add it to the report or something?

Reply:
doc updated